### PR TITLE
rm MLUtils

### PR DESCRIPTION
**Pull request by mcabbott**
This tiny package loads MLUtils only to overload one method of its `batch`, which  ~~ought really to call `Base.stack` anyway.~~ does now call `Base.stack`.

 If we remove that, then we can have this:
```
julia> @time_imports using OneHotArrays
      0.4 ms  Adapt
      0.1 ms  Compat
     82.3 ms  ChainRulesCore
      2.9 ms  GPUArraysCore
      0.2 ms  Requires
     24.3 ms  NNlib 55.87% compilation time (14% recompilation)
     36.9 ms  OneHotArrays
```
instead of this:
```
julia> @time_imports using OneHotArrays
      0.4 ms  Adapt
      0.1 ms  Compat
     85.3 ms  ChainRulesCore
      3.1 ms  CEnum
     17.2 ms  Preferences
      0.3 ms  JLLWrappers
      5.0 ms  LLVMExtra_jll 56.50% compilation time
    115.8 ms  LLVM 48.34% compilation time
      0.2 ms  Reexport
      3.2 ms  GPUArraysCore
    876.6 ms  GPUArrays
      7.7 ms  ShowCases
      1.3 ms  ConstructionBase
     19.2 ms  InitialValues
      0.3 ms  Requires
      0.1 ms  DataValueInterfaces
      1.0 ms  DataAPI
      0.1 ms  IteratorInterfaceExtensions
      0.1 ms  TableTraits
     15.2 ms  Tables
     15.9 ms  MacroTools
      0.2 ms  ZygoteRules
      3.3 ms  StaticArraysCore
     15.8 ms  Setfield
     45.1 ms  BangBang 46.64% compilation time (10% recompilation)
      1.3 ms  ContextVariablesX
      0.1 ms  FLoopsBase
      0.9 ms  PrettyPrint
      0.3 ms  NameResolution
     21.4 ms  MLStyle
      3.5 ms  JuliaVariables
      0.4 ms  ArgCheck
     24.0 ms  Baselet
      0.2 ms  CompositionsBase
      0.1 ms  DefineSingletons
     10.1 ms  MicroCollections
      8.6 ms  SplittablesBase
     62.4 ms  Transducers 23.84% compilation time
      7.1 ms  FLoops
      1.1 ms  InverseFunctions
     26.5 ms  Accessors 24.23% compilation time
     13.6 ms  FunctionWrappers
    463.8 ms  FoldsThreads 91.08% compilation time
     92.7 ms  DataStructures
      0.3 ms  SortingAlgorithms
      9.1 ms  Missings
      3.7 ms  DocStringExtensions 70.13% compilation time
      0.9 ms  ChangesOfVariables
      6.4 ms  IrrationalConstants
      0.7 ms  LogExpFunctions
      0.3 ms  StatsAPI
     33.3 ms  StatsBase
      0.9 ms  DelimitedFiles
      6.6 ms  MLUtils
     23.7 ms  NNlib 49.30% compilation time
     44.4 ms  OneHotArrays
```

### CODE REVIEW

File: src/array.jl [hunk 1]
```diff
@@ -129,7 +129,16 @@ Base.hcat(x::OneHotMatrix, xs::OneHotMatrix...) =
 Base.hcat(x::OneHotVector, xs::OneHotVector...) =
   OneHotMatrix(reduce(vcat, _indices.(xs); init = _indices(x)), _nlabels(x, xs...))
 
-MLUtils.batch(xs::AbstractArray{<:OneHotVector}) = OneHotMatrix(_indices.(xs), _nlabels(xs...))
+if isdefined(Base, :stack)
+  import Base: _stack
+else
+  import Compat: _stack
+end
+function _stack(::Colon, xs::AbstractArray{<:OneHotArray})
```
Comment (CarloLucibello) [inline src/array.jl:137]: Why overload an internal method?

Comment (mcabbott) [reply src/array.jl:137]: It means we can dispatch on `dims`, to catch just this case.

Comment (CarloLucibello) [reply src/array.jl:137]: Could we just error out if `dims != :` instead?

Comment (mcabbott) [reply src/array.jl:137]: But those are valid operations, it should just make an Array as usual.

We could get that by testing and then calling the internal method. Or testing and then `@invoke`. This just seemed simpler.

**Review by CarloLucibello — APPROVED** [review]

**Pull request merged**